tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptions/SubscriptionsControllerUpdatePrepaidSubscriptionConfigurationTest.java

```diff
@@ -0,0 +1,169 @@
+import com.maxio.advancedbilling.models.Subscription;
+import com.maxio.advancedbilling.models.UpsertPrepaidConfiguration;
+import com.maxio.advancedbilling.models.UpsertPrepaidConfigurationRequest;
+import com.maxio.advancedbilling.utils.TestSetup;
+import com.maxio.advancedbilling.utils.TestTeardown;
+import com.maxio.advancedbilling.utils.assertions.ApiExceptionAssert;
+import com.maxio.advancedbilling.utils.assertions.CommonAssertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static com.maxio.advancedbilling.utils.assertions.CommonAssertions.assertUnauthorized;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+public class SubscriptionsControllerUpdatePrepaidSubscriptionConfigurationTest {
+    private static final TestSetup TEST_SETUP = new TestSetup();
+    private static final AdvancedBillingClient CLIENT = TestClient.createClient();
+    private static final SubscriptionsController SUBSCRIPTIONS_CONTROLLER = CLIENT.getSubscriptionsController();
+
+    private static Product product;
+    private static Customer customer;
+
+    @BeforeAll
+    static void setUp() throws IOException, ApiException {
+        product = TEST_SETUP.createProduct(TEST_SETUP.createProductFamily());
+        customer = TEST_SETUP.createCustomer();
+    }
+
+    @AfterAll
+    static void teardown() throws IOException, ApiException {
+        new TestTeardown().deleteCustomer(customer);
+    }
+
+    @Test
+    void shouldThrow404IfSubscriptionDoesNotExist() {
+        // when then
+        CommonAssertions.assertNotFound(() -> SUBSCRIPTIONS_CONTROLLER
+                .updatePrepaidSubscriptionConfiguration(123, new UpsertPrepaidConfigurationRequest()));
+    }
+
+    @Test
+    void shouldReturn422WhenRequestBodyIsEmpty() throws IOException, ApiException {
+        // given
+        Subscription subscription = TEST_SETUP.createSubscription(
+                customer,
+                product,
+                subscriptionCustomizer -> subscriptionCustomizer
+                        .paymentCollectionMethod(CollectionMethod.PREPAID)
+                        .prepaidConfiguration(new UpsertPrepaidConfiguration.Builder()
+                                .initialFundingAmountInCents(1000L)
+                                .build()
+                        )
+        );
+
+        // when - then
+        new ApiExceptionAssert<>(() -> SUBSCRIPTIONS_CONTROLLER.updatePrepaidSubscriptionConfiguration(
+                subscription.getId(),
+                new UpsertPrepaidConfigurationRequest()
+        ))
+                .hasErrorCode(422)
+                .hasMessage("HTTP Response Not OK. Status code: 422. Response: '{errors:{prepaid_configuration:can't be blank}}'.");
```

Comment:
Use assertion for specific Error model to validate it's deserialized correctly.

Reply:
I think it's this generic ApiException, because sdk doesn't support oneOf in exceptions

Reply:
Exactly 👍